aten/src/ATen/EmptyTensor.cpp

```diff
@@ -163,6 +184,27 @@ TensorBase empty_strided_generic(
   return tensor;
 }
 
+TensorBase empty_strided_symint_generic(
+    SymIntArrayRef size,
+    SymIntArrayRef stride,
+    c10::Allocator* allocator,
+    c10::DispatchKeySet ks,
+    ScalarType scalar_type) {
+  at::detail::raise_warning_for_complex_half(scalar_type);
+  caffe2::TypeMeta dtype = scalarTypeToTypeMeta(scalar_type);
+  SymInt size_bytes = computeStorageNbytesSymInt(size, stride, dtype.itemsize());
+  auto storage_impl = c10::make_intrusive<StorageImpl>(
+      c10::StorageImpl::use_byte_size_t(),
+      size_bytes,
+      allocator,
+      /*resizeable=*/true);
+
+  auto tensor = detail::make_tensor_base<TensorImpl>(
+      std::move(storage_impl), ks, dtype);
+  tensor.unsafeGetTensorImpl()->set_sym_sizes_and_strides(size, stride);
+  return tensor;
+}
```

Comment:
@bdhirsh what do you feel about just converting these functions into templates. Reaaally trying to avoid duping them

Reply:
Agreed that the code duping feels bad.

Are you thinking of fully converting the headers into templates, like `at::detail::empty_strided_meta<IntArrayRef>`? (removing all the duplication at the cost of the API requiring templates).

Or keeping the API keeping `empty_strided_meta` + `empty_strided_symint_meta` around, and just internally wrapping their implementation in a shared template. (which... would cut down on some duplication, but I most of the LoC add here just comes from the extra signatures being added.)

Just want to confirm because the first one probably involves updating a bunch of call-sites :p.

Also, are you suggesting to rename `set_sym_sizes_and_strides(SymIntArrayRef sizes, ...)` to `set_sizes_and_strides(SymIntArrayRef sizes, ...)`? Otherwise I'm not sure how we can really de-dup this particular function (`empty_strided_symint_generic`).

Reply:
Latter.